packages/client/src/base/editor-context-service.ts

```diff
@@ -196,6 +205,21 @@ export class EditorContextService implements IActionHandler, Disposable, IDiagra
         this.onModelRootChangedEmitter.fire(root);
     }
 
+    /**
+     * Notifies the service about a viewport change. This method should not be called
+     * directly. It is called by the `CommandStack` after a viewport modifying command has been processed.
+     * @throws an error if the notifier is not a `CommandStack`
+     * @param viewport the new viewport
+     * @param notifier the object that triggered the viewport change
+     */
+    notifyViewportChanged(viewport: Readonly<Viewport>, notifier: AnyObject): void {
```

Comment:
Effectively what we call this with is `GModelRoot & Viewport`, i.e., the result of the `isViewport` typecheck. Since this is only on the client-side, I wonder whether we shouldn't use that type for the listeners instead of just the `Viewport`? Alternatively, we can call this method with just `this.currentViewport` in the command stack.

Reply:
Fine with me, we can change the type of the event and fire it with this.currentViewport.